assets/app/scripts/services/labelFilter.js

```diff
@@ -0,0 +1,337 @@
+
+    $('.label-filter-add').click(function() {
+      // grab the values before clearing out the fields
+      var key = $('.label-filter-key')[0].selectize.getValue();
+      var operator = $('.label-filter-operator')[0].selectize.getValue();
+      var values = $('.label-filter-values')[0].selectize.getValue();
+
+      $('.label-filter-key')[0].selectize.clear();
+      $('.selectize-control.label-filter-operator').hide();
+      $('.label-filter-operator')[0].selectize.clear();
+      $('.selectize-control.label-filter-values').hide();
+      $('.label-filter-values')[0].selectize.clear();
+      $('.label-filter-add').addClass("disabled").prop('disabled', true);              
+
+      // show the filtering active indicator and add the individual filter to the list of active filters
+      $('.label-filter-active').show();
+      self._addActiveFilter({key: key, operator: operator, values: values});
+    });
+
+    $('.label-filter-active').click(function() {
+      $('.label-filter-active').hide();
+      $('.label-filter-active-filters').empty();
+      self._clearActiveFilters();
+    }); 
+
+    // If we are transitioning scenes we may still have filters active but be re-creating the DOM for the widget
+    if (!$.isEmptyObject(this._activeFilters)) {
+      $('.label-filter-active').show();
+      $each(this._activeFilters, function(id, filter) {
+        self._renderActiveFilter(filter);
+      })
+    }   
+  };
+
+  LabelFilter.prototype._getLabelForFilter = function(filterDetails) {
+    var filterLabel = filterDetails.key;
+    if (filterDetails.operator != "EXISTS") {
+      if (filterDetails.operator == "NOT_IN") {
+        filterLabel += " not";
+      }
+      filterLabel += " in (";
+      for (var i = 0; i < filterDetails.values.length; i++) {
+        filterLabel += filterDetails.values[i];
+        if (i != filterDetails.values.length - 1) {
+          filterLabel += ", ";
+        }
+      }
+      filterLabel += ")";
+    }
+    return filterLabel;
+  };
+
+  LabelFilter.prototype._getIdForFilter = function(filter) {
+    return filter.key + "-" + filter.operator + "-" + filter.values.join("-");
+  };
+
+  LabelFilter.prototype._addActiveFilter = function(filter) {
+    this._activeFilters[this._getIdForFilter(filter)] = filter;
+    this._onActiveFiltersChangedCallbacks.fire(this._activeFilters);  
+    this._renderActiveFilter(filter);
+  };
+
+  LabelFilter.prototype._renderActiveFilter = function(filter) {
+    // render the new filter indicator
+    $('<a>')
+      .addClass("label label-default label-filter-active-filter")
+      .prop("href", "javascript:;")
+      .prop("filter-label-id", this._getIdForFilter(filter))
+      .click($.proxy(this, '_removeActiveFilter'))
+      .append(
+        $('<span>')
+          .text(this._getLabelForFilter(filter))
+          // TODO move to the less styles instead
+          .css("padding-right", "5px")
+      )
+      .append(
+        $('<i>')
```

Comment:
see the font-awesome examples http://fortawesome.github.io/Font-Awesome/examples/

not necessary to be an <i> but a common shorthand